Feat: 온보딩 프로세스 구현, 로그인 네이티브 구현

south-nova opened the pull request:
<!-- 이 PR을 요약한 내용으로 위 제목 폼을 채워 주세요. -->
![PR-banner](https://github.com/user-attachments/assets/e9a3d5a0-d8ab-4486-ac39-38b59c211172)

## 작업 개요

[KDDEV-129 온보딩 페이지 구현](https://jira-knockdog.atlassian.net/browse/KDDEV-127)

## 작업 내용
- 소셜 로그인 구현 (`Kakao`, `Google`)
- 온보딩 프로세스 구현
- 네이티브 스택 `Link` 컴포넌트 `StackLink` 추가
- `react-hook-form`, `@hookform/resolvers`, `framer-motion` 추가
- `app.json` -> `app.config.ts` 변경 (환경변수 사용 위함)

---

@im-na0 @yeomgahui @bbb4756

south-nova committed:
feat: 주소 검색 API 호출 메서드 및 useQuery 훅 추가

south-nova committed:
feat: 주소 하이라이트 텍스트 유틸리티 함수 추가

south-nova committed:
feat: 주소 검색 모델 추가

south-nova committed:
feat: 주소 검색 비지니스 hook 구현

south-nova committed:
feat: AddressPicker 컴포넌트 UI 추가

south-nova committed:
refactor: LocationField 비즈니스 로직, 추가/삭제 버튼 기능 추가

south-nova committed:
chore: @babel+core@7.28.4로 업데이트

south-nova committed:
chore: @hookform/resolvers@5.2.2 및 react-hook-form@7.66.0 추가, zod 버전 4.1.12로 업데이트

south-nova committed:
chore: UI 패키지에 React 라이브러리 추가

south-nova committed:
feat: 장소 등록 및 주소 검색 페이지 구현

south-nova committed:
feat: 사용자 프로필 등록 페이지 구현

south-nova committed:
feat: 돌보는 강아지 여부 페이지 구현

south-nova committed:
feat: 강아지 프로필 관련 컴포넌트 리팩토링

south-nova committed:
fix: 주소 추가 버튼 로직 수정 및 경로 변경

south-nova committed:
feat: 주소 모델 및 타입 정의 추가

south-nova committed:
fix: 사용자 주소 모델에서 alias 및 address 필드 선택적 속성으로 변경

south-nova committed:
feat: StackLink 컴포넌트 추가 및 useStackNavigation에서 PushOptions 타입 내보내기

south-nova committed:
feat: 환영 페이지 컴포넌트 및 라우팅 추가

south-nova committed:
feat: 마케팅 정보 수신 동의 페이지 컴포넌트 추가

south-nova committed:
feat: 반려동물 등록 페이지 컴포넌트 추가 (상세정보, 프로필, 관계)

south-nova committed:
feat: 프로필 등록 및 반려동물 관련 페이지 레이아웃 추가

south-nova committed:
fix: ToggleRoot 컴포넌트의 버튼에 type 속성 추가

south-nova committed:
feat: 레거시 프로필 관련 페이지 및 컴포넌트 삭제

south-nova committed:
feat: framer-motion 패키지 추가 및 의존성 업데이트

south-nova committed:
feat: iOS 프로젝트에 Kakao SDK 및 Google SignIn 통합, Podfile.lock 및 Info.plist 업데이트
- Kakao SDK 및 Google SignIn 관련 의존성 추가
- AppDelegate.swift에서 KakaoTalk 로그인 URL 처리 로직 추가
- Info.plist에 URL 스킴 및 로컬라이제이션 설정 추가
- Podfile.properties.json에 새로운 설정 추가

south-nova committed:
feat: Kakao SDK 및 Google SignIn 초기화 로직 추가
- App.tsx에서 Kakao SDK 및 Google SignIn 초기화 로직 구현
- 환경 변수에서 앱 키 및 클라이언트 ID 로드

south-nova committed:
feat: Field 컴포넌트 및 관련 서브컴포넌트 추가

- Field, FieldLabel, FieldLabelIndicator, FieldDescription, FieldError, FieldGroup, FieldLegend, FieldSeparator, FieldSet, FieldContent, FieldTitle 컴포넌트 구현
- index.ts 파일에 Field 컴포넌트 내보내기 추가
- slot/index.ts 파일 생성 및 Slot 컴포넌트 내보내기 추가

south-nova committed:
feat: 소셜 로그인 메서드 및 결과 타입 추가

- methods.ts 파일에 kakaoLogin, googleLogin, appleLogin 메서드 추가
- SocialLoginResult 타입 정의 및 index.ts에 내보내기 추가

south-nova committed:
feat: 소셜 인증 핸들러 등록 로직 추가

- register-auth.ts 파일 생성 및 Kakao, Google 로그인 핸들러 구현
- register-all.ts 파일에 인증 핸들러 등록 추가

south-nova committed:
feat: 반려동물 상세 페이지 및 API 추가

- 반려동물 상세 페이지를 위한 PetDetailPage 컴포넌트 추가
- 반려동물 모델 및 API 메서드(postRegisterPet, postUpdatePetDetail) 정의
- 반려동물 관련 훅(usePetRegisterMutation, usePetUpdateDetailMutation) 추가
- 반려동물 모델에 대한 타입 정의 및 내보내기 추가

south-nova committed:
feat: 사용자 및 소셜 유저 스토어에 지속성 추가

- useSocialUserStore 및 useUserStore에 Zustand의 persist 미들웨어 적용
- 소셜 유저 및 사용자 정보를 로컬 스토리지에 저장하도록 설정
- 사용자 등록 API(postRegisterUser) 및 관련 훅(useUserRegisterMutation) 추가
- User 인터페이스에 id 필드 추가 및 UserAddress에서 detail 필드 제거

south-nova committed:
feat: tempBridgeLogin 제거 및 소셜 로그인 로직 업데이트

- tempBridgeLogin.ts 파일 삭제
- useLogin 훅에서 tempBridgeLogin 호출을 bridge.request로 변경하여 소셜 로그인 처리 방식 개선
- 로그인 후 네비게이션 로직을 back()으로 수정
- 관련된 경로 변경 및 오류 처리 로직 업데이트

south-nova committed:
feat: RelationshipSelector 컴포넌트 개선 및 자동 포커스 기능 추가

south-nova committed:
feat: 등록 경로 및 페이지 업데이트

- route.ts 파일에서 프로필 경로를 등록 경로로 변경
- 새로운 등록 관련 경로 추가: 장소 등록, 사용자 닉네임, 반려동물 등록 및 상세 페이지
- MarketingConsentPage, PetDetailPage, PetOwnershipPage, PetProfilePage, PetRelationshipPage, UserNicknamePage에서 경로 상수 사용으로 리팩토링
- 각 페이지에서 경로 변경에 따른 네비게이션 로직 업데이트

south-nova committed:
feat: 모바일 앱 의존성 업데이트 및 새로운 패키지 추가

- @react-native-google-signin/google-signin, @react-native-kakao/core, @react-native-kakao/user, expo-apple-authentication, expo-build-properties 패키지 추가
- pnpm-lock.yaml 파일에서 관련 의존성 버전 업데이트

south-nova committed:
feat: tsconfig.json 파일 업데이트

- TypeScript 경로 설정을 배열 형식으로 변경하여 가독성 향상
- .expo/types/**/*.ts 및 expo-env.d.ts 파일 포함 추가

south-nova committed:
feat: 스토리지 상수에 사용자 관련 키 추가

- USER 및 SOCIAL_USER 키를 STORAGE_KEYS에 추가하여 사용자 및 소셜 사용자 정보를 저장할 수 있도록 개선

south-nova committed:
chore: 사용하지 않는 페이지 삭제

south-nova committed:
fix: useStackNavigation 내보내기 수정

- useStackNavigation에서 PushOptions 타입 제거하여 불필요한 내보내기 정리

south-nova committed:
fix: useLocationField에서 경로 상수 사용으로 리팩토링

- useLocationField 훅에서 하드코딩된 경로를 route 상수로 변경하여 유지보수성 향상

south-nova committed:
fix: RedirectLoginPage에서 useStackNavigation 및 상태 관리 개선

- useRouter를 useStackNavigation으로 변경하여 네비게이션 로직 개선
- linkedSocialUser 상태를 useState로 관리하고, useEffect를 통해 스토리지에서 사용자 정보를 가져오도록 수정
- linkedSocialUser가 없을 경우 null 반환으로 렌더링 로직 개선

south-nova committed:
feat: LocationRegisterPage에서 주소 등록 로직 개선

- useForm 훅에 useWatch 추가하여 홈 주소 입력 여부 확인
- 사용자 등록 시 주소 데이터를 필터링하고 별칭 추가하는 로직 구현
- ActionButton의 비활성화 상태를 홈 주소 입력 여부에 따라 설정하여 사용자 경험 향상

south-nova committed:
fix: Header 컴포넌트 스타일 개선

south-nova committed:
Merge branch 'develop' into KDDEV-129-feat/onboarding

yeomgahui commented on apps/knockdog/src/entities/address/model/address.ts:
```diff
@@ -1,3 +1,28 @@
+<<<<<<< HEAD
```
성남님 여기 충돌 코드가 들어간것 같아여

south-nova replied:
넵 수정했습니다!

south-nova committed:
fix: address 충돌 픽스

south-nova committed:
fix: BreedSelector, GenderSelector, NeuteredSelector, YearSelector 컴포넌트 내보내기 방식 수정

south-nova committed:
feat: 반려동물 등록 페이지에 세부 정보 입력 필드 및 훅 추가

- petDetailFields.tsx 파일 생성하여 반려동물 세부 정보 입력 필드 정의
- usePetDetail.ts 파일 생성하여 반려동물 세부 정보 관리 로직 구현
- PetDetailPage.tsx에서 기존 로직을 usePetDetail 훅으로 리팩토링하여 코드 간결화 및 재사용성 향상

south-nova committed:
fix: PetRelationshipPage에서 쿼리 파라미터 수정

- petId와 petName만을 쿼리로 전달하도록 수정하여 불필요한 relation 파라미터 제거

south-nova committed:
fix: postUpdatePetDetail 함수에서 API 응답 타입 지정

- postUpdatePetDetail 함수에 .json<ApiResponse<PetResponse>>() 추가하여 API 응답의 타입을 명시적으로 지정하여 타입 안정성 향상

south-nova committed:
fix: usePetUpdateDetailMutation export 추가

south-nova committed:
fix: renderHighlightText 키워드 하이라이트 로직 개선

south-nova committed:
fix: react-dom 버전 범위 수정

south-nova committed:
fix: pnpm-lock.yaml 및 package.json에서 react 및 react-dom 버전 범위 수정
